paper/paper.md

```diff
@@ -0,0 +1,175 @@
+obviating the need to maintain two separate versions. nbgrader also
+automatically grades submitted assignments by executing the notebooks and
+storing the results of the tests in a database. After auto-grading, instructors
+can manually grade free responses and provide partial credit using the
+*formgrader* notebook extension. Finally, instructors can use nbgrader to leave
+personalized feedback for each student's submission, including comments as well
+as detailed error information.
+
+nbgrader can also be used with JupyterHub [@JupyterHub], which is a centralized,
+server-based installation that manages user logins and management of Jupyter
+notebooks. When used with JupyterHub, nbgrader provides additional workflow
+functionality, covering the entire grading process. After creating an
+assignment, instructors can distribute it to students, who can then fetch a copy
+of the assignment directly through the notebook server interface. Students can
+submit their completed assignment through the same interface, making it
+available for instructors. After students submit their assignments, instructors
+can collect the assignments with a single command and use the auto-grading
+functionality in the normal way.
+
+Since its conception in September 2014, nbgrader has been used in a number of
+educational contexts, including courses at UC Berkeley, Cal Poly, University of
+Pretoria, University of Edinburgh, Northeastern University, Central Connecticut
+State University, KTH Royal Institute of Technology, CU Boulder, University of
+Amsterdam, George Washington University, Texas A&M, Bryn Mawr College, and
+University of Maryland; and, as of May 2018, over 10,000 nbgrader-based
+notebooks exist on GitHub. In addition to its core functionality, nbgrader has
+expanded to support a number of other features, including the ability to handle
+multiple courses on the same JupyterHub instance; the option to either include
+or hide automatically graded tests; customizable late penalties; and support for
+importing assignment files downloaded from a Learning Management System (LMS).
+
+# Statement of Need
+
+The use of computational methods has become increasingly widespread in fields
+outside of computer science [@wing2008computational]. As these disciplines begin
+to require computational tools, undergraduate curricula also begin to include
+topics in programming and computer science. However, perhaps because students
+are focused on the discipline that is the object of their study—and programming
+is likely a secondary interest—it has been shown that teaching computer science
+can be more effective when courses include interdisciplinary motivations
+[@Cortina2007; @Forte2005; @Guzdial2005]. One approach for teaching programming
+in a way that facilitates exploration with interdisciplinary questions is to
+teach students computational concepts in an interactive environment where it is
+possible to quickly write, test, and tweak small units of code. Many such
```

Comment:
This is really great. Anecdotally, it seems that there's a feedback loop here that strengthens students' interest in programming as a discipline when programming helps them understand course concepts in the context of a reading or homework assignment. Nbgrader's facilitation of distributing assignments en masse allows instructors to encourage students to treat programming like a practice. One or two "programming projects" during a semester has nowhere near the effect of small, daily assignments that can be graded quickly. (There's no action item here. I think the prose is really tight)

Reply:
Thanks!